docs/proposals/removal-tekton-resources.md

```diff
@@ -0,0 +1,269 @@
+
+The results for the sources will be filled by our own code as described later. A build strategy author must not do anything to fill them.
+
+For the output image, we will also provide two more system-defined results: `image-digest` and `image-size`. As the image building process is part of the build strategy, only the build strategy author can have the knowledge about where to get those values from and subsequently to write them to files. The build strategy author will access the path for the result files using Tekton's replacement variables for results, in this case: `$(results.image-digest.path)` and `$(results.image-size.path)`. This will be documented in our build strategy documentation. A build strategy author is not required to provide a value for those results. In case values are not provided, they will not be present in the TaskRun status.
+
+In our sample build strategies, we should support these two results as much as possible:
+
+- `ko` supports to write an OCI image manifest file using the `--oci-layout-path` argument. The strategy needs to write that manifest to a temporary location, use a tool to extract the digest and size from the index.json file, and write this data to the result files.
+- Kaniko supports to write an OCI image manifest file using the `--oci-layout-path` argument. Kaniko is based on scratch, as such, we cannot extend the existing build-and-push step to use some tool and run some extraction logic. We need to add an extra step. For this, we need to add volumeMounts to write the temporary file in the `build-and-push` step, and consume it in the following one to extract and store the results.
+- The strategy that uses s2i in combination with Kaniko needs to adopt the solution for Kaniko.
+- The Buildpacks strategies can use the `--report` argument for the [create](https://buildpacks.io/docs/concepts/components/lifecycle/create/) command to write a toml file. From there, the value can be extracted using shell logic like in the [Tekton catalog](https://github.com/tektoncd/catalog/blob/main/task/buildpacks/0.3/buildpacks.yaml#L153). Though, I prefer to use the same step to save resources. As far as I remember, only the digest is available but no size.
+- Buildah TBD `buildah inspect` ?
+- BuildKit writes the images digest to stdout only atm. There is a long-standing issue, [Add some way to get the sha256 digest of the image #1158](https://github.com/moby/buildkit/issues/1158), but no resolution yet. One could use a tool (like [`crane`](https://github.com/google/go-containerregistry/blob/main/cmd/crane/doc/crane.md)) that checks the metadata of the image after it was pushed. (We could also chose to generally use this approach generally and make it part of the Shipwright logic. It would just not be efficient in many cases where the information is locally available already.)
```

Comment:
In all of these cases, we could run [`crane digest`](https://github.com/google/go-containerregistry/blob/main/cmd/crane/doc/crane_digest.md) (or similar, or build our own logic) to fetch the digest of the image that was just pushed to the registry. There's some chance of a race condition (as in #731), but it's better than nothing.

Reply:
Nice, I missed that command. I thought it would be `crane manifest` and then extract the digest with `jq`, but this command makes it very simple. I might have asked already, but is there an officially supported image with `crane` included?

Reply:
Yep, https://gcr.io/go-containerregistry/crane